packages/atomic/src/components/facets-v1/facet-search/facet-search-utils.ts

```diff
@@ -0,0 +1,39 @@
+import {FacetSearchState} from '@coveo/headless';
+
+/**
+ * Meant to be used inside the `componentShouldUpdate` lifecycle method.
+ * It prevents updating the facet between two matchless facet searches.
+ * It also prevents updating the facet until the first search has loaded.
+ */
+export function shouldUpdateFacetSearchComponent(
+  nextFacetSearchState: FacetSearchState,
```

Comment:
Might be just me, but it's reversed :D 

`shouldUpdateFacetSearchComponent(prev/current, next)`

Reply:
That's how stencil does it ,for componentShouldUpdate, so I kept the same order 🤷‍♂️ 
https://stenciljs.com/docs/component-lifecycle#componentshouldupdate
> This hook receives three arguments: the new value, the old value and the name of the changed state